fineract-provider/src/main/java/org/apache/fineract/cob/service/LoanAccountLockServiceImpl.java

```diff
@@ -51,4 +51,11 @@ public boolean isLoanHardLocked(Long loanId) {
                         || LockOwner.LOAN_INLINE_COB_PROCESSING.equals(loanAccountLock.getLockOwner())) //
                 .filter(loanAccountLock -> StringUtils.isBlank(loanAccountLock.getError())).isPresent();
     }
+
+    @Override
+    public boolean isLoanSoftLocked(Long loanId) {
+        Optional<LoanAccountLock> loanAccountLockOptional = loanAccountLockRepository.findById(loanId);
```

Comment:
I think it's not needed to fetch the actual entity. Let's write a custom repository method for this.
Something like `existsByIdAndLockOwner`